### ISSUE TITLE
Cisco_vlan - proper state of VLAN name is not captured

### ISSUE THREAD

**Issue by tomcooperca** [cisco/cisco-network-puppet-module#408]
Hi,
ciscopuppet module version 1.4.1
cisco_node_utils 1.4.1
Puppet server version 4.7
Agent version 4.5.2
NX-OS 7.0(3)I4(2)

When applying cisco_vlan resources, the agent on the switch is unable to proper gather state of the name and causes the resource to fail due to "VLAN with the same name exists" CLI error. When looking at the debug output of the Puppet run, the name of the VLAN is being trimmed down to 31 characters even though the switch (N9K-C9396PX) is using "system vlan long-name". It looks to be caused by the CLI command being used to gather the state of the VLAN. "show vlan brief" truncates the full VLAN name due the CLI output formatting. The VLAN name column on the CLI is only 31 characters long and any additional characters are put on the next line in the output. The workaround is to use a shorter VLAN name but it would be nice to have up to 128 characters fully supported via Puppet. 
See below:

```
pod2-switch1# show vlan brief

VLAN Name                             Status    Ports
---- -------------------------------- --------- -------------------------------
1    default                          active    Po100, Eth1/4, Eth1/5, Eth1/6
                                                Eth1/7, Eth1/8, Eth1/9, Eth1/10
                                                Eth1/11, Eth1/12, Eth1/13
                                                Eth1/14, Eth1/15, Eth1/16
                                                Eth1/17, Eth1/18, Eth1/19
                                                Eth1/20, Eth1/21, Eth1/22
                                                Eth1/23, Eth1/24, Eth1/25
                                                Eth1/26, Eth1/27, Eth1/28
                                                Eth1/29, Eth1/30, Eth1/31
                                                Eth1/32, Eth1/33, Eth1/34
                                                Eth1/35, Eth1/36, Eth1/37
                                                Eth1/38, Eth1/39, Eth1/40
                                                Eth1/41, Eth1/42, Eth1/43
                                                Eth1/44, Eth1/45, Eth1/46
                                                Eth1/47, Eth1/48, Eth2/5, Eth2/6
                                                Eth2/7, Eth2/8, Eth2/9, Eth2/10
                                                Eth2/11, Eth2/12
2    PTP,UNDERLAY,PEER-GW             active    Po100, Eth1/47, Eth1/48
10   puppet_L3VNI                     active    Po100, Eth1/47, Eth1/48
505  puppet_10.0.105.208/29_VLAN505   active    Po3, Po100, Eth1/3, Eth1/47
                                                Eth1/48
1001 puppet_10.0.101.0/24_VLAN1001    active    Po3, Po100, Eth1/3, Eth1/47
                                                Eth1/48
2102 puppet_10.0.102.128/25_DHCPVLAN2 active    Po1, Po2, Po100, Eth1/1, Eth1/2
     102                                        Eth1/47, Eth1/48
2103 puppet_10.0.103.64/26_DHCPVLAN21 active    Po1, Po2, Po100, Eth1/1, Eth1/2
     03                                         Eth1/47, Eth1/48
3104 puppet_10.0.104.96/27_DHCPVLAN31 active    Po100, Eth1/47, Eth1/48
     04         
```                                

Debug: Get state using data format 'cli'
Debug:   executing command:
    show vlan brief
Debug:   to get value:     /^2102\s+\S+\s+(\S+)\s/
Debug: Input (cli_show_ascii): 'show vlan brief'
Debug: Massaging '["active"]' (["active"])
Notice: /Stage[main]/Ciscopuppet::Vlans/Cisco_vlan[2102]/vlan_name: vlan_name changed '**puppet_10.0.102.128/25_DHCPVLAN2**' to '**puppet_10.0.102.128/25_DHCPVLAN2102**'
Debug: Set state using data format 'cli'
Debug:   with context:
    vlan 2102
Debug:   to value(s):
     name puppet_10.0.102.128/25_DHCPVLAN2102 ; end
Debug: Input (cli_conf): 'vlan 2102 ;  name puppet_10.0.102.128/25_DHCPVLAN2102 ; end'
Debug: Sending HTTP request to NX-API at localhost:
{"accept-encoding"=>["gzip;q=1.0,deflate;q=0.6,identity;q=0.3"], "accept"=>["*/*"], "user-agent"=>["Ruby"], "cookie"=>["nxapi_auth=admin:local"], "content-type"=>["application/json"]}
{"ins_api":{"version":"1.0","type":"cli_conf","chunk":"0","sid":"1","input":"vlan 2102 ;  name puppet_10.0.102.128/25_DHCPVLAN2102 ; end","output_format":"json"}}
Debug: HTTP Response: OK
{
        "ins_api":      {
                "sid":  "eoc",
                "type": "cli_conf",
                "version":      "1.0",
                "outputs":      {
                        "output":       [{
                                        "code": "200",
                                        "msg":  "Success",
                                        "body": {
                                }
                                }, {
                                        "code": "200",
                                        "msg":  "Success",
                                        "body": {
                                }
                                }, {
                                        "body": "**ERROR: VLAN with the same name exists - vlan puppet_10.0.102.128/25_DHCPVLAN2102\nCannot run commands in the mode at this moment. Please try again.\n**",
                                        "code": "200",
                                        "msg":  "Success"
                                }]
                }
        }
}
Debug: Result for 'vlan 2102': Success
Debug: Result for ' name puppet_10.0.102.128/25_DHCPVLAN2102 ; end': Success
Debug: Result for '': Success
Debug: Output: ERROR: VLAN with the same name exists - vlan puppet_10.0.102.128/25_DHCPVLAN2102
Cannot run commands in the mode at this moment. Please try again.

Error: /Stage[main]/Ciscopuppet::Vlans/Cisco_vlan[2102]: Could not evaluate: ERROR: VLAN with the same name exists - vlan puppet_10.0.102.128/25_DHCPVLAN2102
Cannot run commands in the mode at this moment. Please try again.

**Comment by mikewiebe** [comment]
@tomcooperca, Thanks for making us aware of this issue.  We are currently investigating potential fixes.

**Comment by mikewiebe** [comment]
@tomcooperca I have a candidate fix for this issue in code review now. https://github.com/cisco/cisco-network-node-utils/pull/505.  We plan to release this fix with the `1.5.0` puppet module.

**Comment by tomcooperca** [comment]
Thanks @mikewiebe!
